packages/@sanity/preview/src/prepareForPreview.js

```diff
@@ -1,74 +1,243 @@
-import {get, pick, debounce, flatten, uniqBy} from 'lodash'
+// @flow
+import {debounce, flatten, get, isPlainObject, pick, uniqBy} from 'lodash'
+import {INVALID_PREVIEW_CONFIG} from './constants'
 
-const pass = v => v
+const identity = v => v
 const PRESERVE_KEYS = ['_id', '_type', '_upload']
+const EMPTY = []
 
-let COLLECTED_ERRORS = {}
+type __DEV__ = boolean
+
+type ViewOptions = {}
+
+type SelectedValue = {}
+
+type PreparedValue = {
+  title: string,
+  subtitle: string,
+  description: string
+}
+
+type PreviewConfig = {
+  select: {[string]: string},
+  prepare?: (SelectedValue, ViewOptions) => PreparedValue
+}
+
+type Type = {
+  name: string,
+  preview: PreviewConfig
+}
+
+type PrepareInvocationResult = {|
```

Comment:
Not familiar with this pipe syntax, what does it mean/do?

Reply:
By default in Flow, object types are open ended, meaning you may keep additional props on it, the pipe is the syntax for "exact type", meaning you'll get a type error if the value contains additional properties.